models/migrations/v210.go

```diff
@@ -0,0 +1,26 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package migrations
+
+import (
+	"fmt"
+
+	"xorm.io/xorm"
+)
+
+func createForeignReferenceTable(x *xorm.Engine) error {
+	type ForeignReference struct {
+		// RepoID is the first column in all indices. now we only need 2 indices: (repo, local) and (repo, foreign, type)
+		RepoID       int64  `xorm:"UNIQUE(repo_foreign_type) INDEX(repo_local)" `
+		LocalIndex   int64  `xorm:"INDEX(repo_local)"` // the resource key inside Gitea, it can be IssueIndex, or some model ID.
+		ForeignIndex string `xorm:"INDEX UNIQUE(repo_foreign_type)"`
+		Type         string `xorm:"VARCHAR(8) INDEX UNIQUE(repo_foreign_type)"`
```

Comment:
```suggestion
		Type         string `xorm:"VARCHAR(16) INDEX UNIQUE(repo_foreign_type)"`
```
Current struct in `foreignreference.go` also uses 16.

Reply:
Good catch! Fixed.

---
[source](https://lab.fedeproxy.eu/forgefriends/forgefriends/-/merge_requests/36#note_1095)